src/components/common/RedactedName/RedactedName.tsx

```diff
@@ -0,0 +1,48 @@
+import { Button, Tooltip, TooltipTrigger } from "@/components/common";
+import { Eye, EyeOff } from "lucide-react";
+import { useState } from "react";
+import type { JSX } from "react/jsx-runtime";
+import { tv } from "tailwind-variants";
+
+export interface RedactedNameProps {
+  children: string;
+  className?: string;
+}
+
+const redactedStyles = tv({
+  base: "inline-flex items-center gap-1.5 rounded-md transition-colors",
+  variants: {
+    isRevealed: {
+      true: "bg-graya-2 dark:bg-graydarka-2",
+      false: "bg-graya-3 dark:bg-graydarka-3 text-transparent select-none",
+    },
+  },
+});
+
+export function RedactedName({
+  children,
+  className,
+}: RedactedNameProps): JSX.Element | null {
```

Comment:
```suggestion
}: RedactedNameProps) {
```
nit: TypeScript should be able to infer this!